Sources/PointFree/Account/Cancel.swift

```diff
@@ -74,60 +111,28 @@ func requireSubscription<A>(
         }
         ?? pure(nil)
 
-      return subscription.flatMap { sub in
-        guard let sub = sub else {
-          return conn
-            |> writeStatus(.notFound)
-            >-> respond(text: "Not subscribed :(")
-        }
-        return conn.map(const(.init(first: sub, second: .init(first: currentUser, second: rest))))
-          |> middleware
-      }
+      return subscription.flatMap { conn.map(const($0 .*. conn.data)) |> middleware }
     }
 }
 
-func requireSubscriptionOwner<A>(
-  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data>
+private func isSubscriptionOwner<A>(_ subscriptionAndUser: T3<Database.Subscription, Database.User, A>) -> Bool {
+  return get1(subscriptionAndUser).userId == get2(subscriptionAndUser).id
+}
+
+private func fetchStripeSubscription<A>(
+  _ middleware: (@escaping Middleware<StatusLineOpen, ResponseEnded, T2<Stripe.Subscription?, A>, Data>)
   )
-  -> Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data> {
+  -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.Subscription, A>, Data> {
 
     return { conn in
-      guard conn.data.first.userId == conn.data.second.first.id else {
-        return conn
-          |> writeStatus(.notFound)
-          >-> respond(text: "Not the subscription owner :(")
-      }
-
-      return conn |> middleware
+      AppEnvironment.current.stripe.fetchSubscription(conn.data.first.stripeSubscriptionId)
```

Comment:
I made a bunch of these fetch transformers that never failover. Is that the right way to go about preparing optional data?